app/dashboard/src/layouts/CategorySwitcher/Category.ts

```diff
@@ -1,34 +1,91 @@
 /** @file The categories available in the category switcher. */
+import type * as backend from '#/services/Backend'
+
+// ====================
+// === CategoryType ===
+// ====================
+
+/** A category with a special predefined meaning. */
+interface StaticCategory {
+  readonly type: Exclude<
+    CategoryType,
+    CategoryType.localDirectory | CategoryType.team | CategoryType.user
+  >
+}
+
+/** A category corresponding to the root directory of a user. */
+interface UserCategory {
+  readonly type: CategoryType.user
+  readonly rootPath: backend.Path
+  readonly homeDirectoryId: backend.DirectoryId
+}
+
+/** A category corresponding to the root directory of a team within an organization. */
+interface TeamCategory {
+  readonly type: CategoryType.team
+  readonly team: backend.UserGroupInfo
+  readonly rootPath: backend.Path
+  readonly homeDirectoryId: backend.DirectoryId
+}
+
+/** A category corresponding to an alternate local root directory. */
+interface LocalDirectoryCategory {
+  readonly type: CategoryType.localDirectory
+  readonly rootPath: backend.Path
+  readonly homeDirectoryId: backend.DirectoryId
+}
+
+/** A category of an arbitrary type. */
+type Category = LocalDirectoryCategory | StaticCategory | TeamCategory | UserCategory
+export default Category
 
 // ================
 // === Category ===
 // ================
 
 /** The categories available in the category switcher. */
-enum Category {
+export enum CategoryType {
```

Comment:
Maybe it's time to split local categories with the cloud ones?

Reply:
we *could*, but is there an advantage, when we want to handle both local and cloud categories in every component?

Reply:
If we can't merge 2 enums then we shouldn't use enums.

Reply:
again my point is that there is no advantage in splitting this enum in the first place. i have nothing against splitting this, but there is no advantage that i can see, it's not like it lets us simplify any code

Reply:
I see these advantages:

1. Easier to understand if a category is local or cloud
2. Less error prone when it comes to the functions like `isCloud`, is `isLocal`
3. Easier to extend in the future

Reply:
i mean, i switched and now all the category types are inside their respective objects (instead of as a string union). not sure if it's ideal but `isCloud` and `isLocal` are able to use the new object unions, but idk if 1 and 3 have been addressed

(note that i'm not against adding a string union, i'm just not sure how to make sure they're in sync with the object types. but also they don't seem necessary yet so maybe we can leave it for later?)